### ISSUE TITLE
Support updating relationships by name instead of by foreign key

### ISSUE THREAD

**Issue by bmac** [endpoints/endpoints#9]
If I have a foregin key column on my model named `rel_foregin_key` but I want to refer to the relationship by the name `relName`. I should be able to perform PUT, and POST actions on the resouce without knowing `rel_foregin_key` and only using `relName`

Example GET JSON

```js
{ applicants: {
  {
    "id": 40,
    "rel_foregin_key": 8,
    "links": {
      "relName": {
        "type": "relationship",
        "id": 8,
        "href": "/relationship/10"
      },
  }
}
```

**Comment by tkellen** [comment]
According to the json-api spec, this should actually be
```js
PUT {
  "applicants": {
    "id": 40,
    "links": {
      "author": "1"
    }
  }
}
```

Can you support that?  If so, I should be able to have support for this in today.

**Comment by bmac** [comment]
It will require an update to ember-data-endpoints but yes I can do that.

**Comment by tkellen** [comment]
Can this also support hasMany relationships? According to the json-api spec, this would mean sending an array of values rather than a single element. It's also important to note that the array is a full replacement, which means the relation is totally cleared and then re-inflated with the provided values.  An empty array would clear the relation entirely.

The spec does have support for updating them individually, and I would like to support that eventually, but I think this is a good first step.

**Comment by bmac** [comment]
I think it could. What does the spec say about to do when you don't know the values of a hasMany relationship? I'm thinking about a the case of an async relationship where you never fetch the relationship so the client never knows what was in the array to begin with.

**Comment by tkellen** [comment]
I would think you just omit the key in this case.

**Comment by tkellen** [comment]
This will be addressed in #15
